net/include: support ipv6 nexthop over lo ecmp

allow ecmp ipv6 mpls encap over dev lo

Signed-off-by: Dewi Morgan <dewi.morgan@intl.att.com>
---
 include/net/ip6_route.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/net/ip6_route.h
+++ b/include/net/ip6_route.h
@@ -73,6 +73,12 @@ static inline bool rt6_need_strict(const
  */
 static inline bool rt6_qualify_for_ecmp(const struct fib6_info *f6i)
 {
+	if (f6i->fib6_nh->fib_nh_dev &&
+	    (f6i->fib6_nh->fib_nh_dev->flags & IFF_LOOPBACK) &&
+	    (f6i->fib6_nh->nh_common.nhc_lwtstate &&
+	    (f6i->fib6_nh->nh_common.nhc_lwtstate->type == LWTUNNEL_ENCAP_MPLS)))
+		return true;
+
 	/* the RTF_ADDRCONF flag filters out RA's */
 	return !(f6i->fib6_flags & RTF_ADDRCONF) && !f6i->nh &&
 		f6i->fib6_nh->fib_nh_gw_family;
